dev/storage.h

```diff
@@ -142,33 +142,82 @@ namespace sqlite_orm {
+                    auto& columnName = storageColumnInfo.name;
+                    if(storageColumnInfo.hidden) {
+                        columnsToAdd.push_back(&storageColumnInfo);
+                    }
+                }
+            }
             template<class I>
-            void backup_table(sqlite3* db, const I& tableImpl, const std::vector<table_info*>& columnsToIgnore) {
+            void drop_create_with_loss(const I& tImpl, sqlite3* db) {
+
+                try {
+                    this->start_migration();
+
+                    this->drop_table_internal(tImpl.table.name, db);
+                    this->create_table(db, tImpl.table.name, tImpl);
+
+                    this->commit_migration();
+                } catch(std::exception& ex) {
+                    std::string s = ex.what();  // 's' for debugging
+                    this->abort_migration();
+                    throw;
+                }
+            }
+
+            template<class I>
+            void backup_table(sqlite3* db, const I& tableImpl, const std::vector<table_xinfo*>& columnsToIgnore) {
 
                 //  here we copy source table to another with a name with '_backup' suffix, but in case table with such
                 //  a name already exists we append suffix 1, then 2, etc until we find a free name..
                 auto backupTableName = tableImpl.table.name + "_backup";
-                if(tableImpl.table_exists(backupTableName, db)) {
+                if(table_exists(backupTableName, db)) {
                     int suffix = 1;
                     do {
                         std::stringstream stream;
                         stream << suffix;
                         auto anotherBackupTableName = backupTableName + stream.str();
-                        if(!tableImpl.table_exists(anotherBackupTableName, db)) {
+                        if(!table_exists(anotherBackupTableName, db)) {
                             backupTableName = anotherBackupTableName;
                             break;
                         }
                         ++suffix;
                     } while(true);
                 }
+                try {
+                    this->start_migration();
+
+                    this->create_table(db, backupTableName, tableImpl);
+
+                    this->copy_table(db, backupTableName, tableImpl, columnsToIgnore);
 
-                this->create_table(db, backupTableName, tableImpl);
+                    this->drop_table_internal(tableImpl.table.name, db);
 
-                this->copy_table(db, backupTableName, tableImpl, columnsToIgnore);
+                    this->rename_table(db, backupTableName, tableImpl.table.name);
 
-                this->drop_table_internal(tableImpl.table.name, db);
+                    this->commit_migration();
 
-                tableImpl.rename_table(db, backupTableName, tableImpl.table.name);
+                } catch(std::exception& ex) {
```

Comment:
Please catch by const-reference, and e.g. comment out the variable name.

Reply:
done